### PR TITLE
e2e_node: Assign enough time to finish the postStart hook

### LINKED ISSUE #120458: Failure cluster [9339868f...] Containers Lifecycle should not launch second container before PostStart of the first container completed
##### Error text:
```
[FAILED] couldn't find that PostStart-regular-1 ever exited, got 2023-08-29 00:39:12 +0000 UTC 49.060000 regular-1 Starting
2023-08-29 00:39:12 +0000 UTC 49.070000 regular-1 Started
2023-08-29 00:39:12 +0000 UTC 49.070000 regular-1 Delaying
2023-08-29 00:39:14 +0000 UTC 50.970000 PostStart-regular-1 Starting
2023-08-29 00:39:14 +0000 UTC 50.980000 PostStart-regular-1 Started
2023-08-29 00:39:14 +0000 UTC 50.980000 PostStart-regular-1 Delaying
2023-08-29 00:39:14 +0000 UTC 51.070000 regular-1 Exiting
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Starting
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Started
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Delaying
2023-08-29 00:39:15 +0000 UTC 52.440000 regular-2 Exiting
In [It] at: test/e2e_node/container_lifecycle_test.go:377 @ 08/29/23 00:39:18.148

```
#### Recent failures:
[2023/9/6 04:07:01 ci-crio-cgroupv1-node-e2e-conformance](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-conformance/1699152150099136512)
[2023/9/6 01:07:02 ci-crio-cgroupv1-node-e2e-conformance](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-conformance/1699106850928070656)
[2023/9/5 23:07:02 ci-crio-cgroupv1-node-e2e-conformance](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-conformance/1699076651507257344)
[2023/9/5 20:07:03 ci-crio-cgroupv1-node-e2e-conformance](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-conformance/1699031352382328832)
[2023/9/5 13:01:01 ci-crio-cgroupv1-node-e2e-conformance](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-crio-cgroupv1-node-e2e-conformance/1698924144361476096)

/kind failing-test

/sig node

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?
/kind failing-test
/kind flake

#### What this PR does / why we need it:

```
[FAILED] couldn't find that PostStart-regular-1 ever exited, got 2023-08-29 00:39:12 +0000 UTC 49.060000 regular-1 Starting
2023-08-29 00:39:12 +0000 UTC 49.070000 regular-1 Started
2023-08-29 00:39:12 +0000 UTC 49.070000 regular-1 Delaying
2023-08-29 00:39:14 +0000 UTC 50.970000 PostStart-regular-1 Starting
2023-08-29 00:39:14 +0000 UTC 50.980000 PostStart-regular-1 Started
2023-08-29 00:39:14 +0000 UTC 50.980000 PostStart-regular-1 Delaying
2023-08-29 00:39:14 +0000 UTC 51.070000 regular-1 Exiting
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Starting
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Started
2023-08-29 00:39:14 +0000 UTC 51.430000 regular-2 Delaying
2023-08-29 00:39:15 +0000 UTC 52.440000 regular-2 Exiting
In [It] at: test/e2e_node/container_lifecycle_test.go:377 @ 08/29/23 00:39:18.148
```

Looking at the error message, the problem seems that the regular-1 container exits before its postStart hook is finished.

This deflakes the `Containers Lifecycle should not launch second container before PostStart of the first container completed` test by assigning enough time to finish the postStart hook.

#### Which issue(s) this PR fixes:

Ref #120458 

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/container_lifecycle_test.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ var _ = SIGDescribe("[NodeConformance] Containers Lifecycle ", func() {
 						Name:  regular1,
 						Image: busyboxImage,
 						Command: ExecCommand(regular1, execCommand{
-							Delay:    2,
+							Delay:    10,
```
Comment (ffromani) [inline test/e2e_node/container_lifecycle_test.go:239]: pretty much LGTM, but there are 2 questions:

1. how was "10" determined? If it was done by trial and error, fine, but please let's record that - so if and when we revisit in the future we have a data point to guide our future decisions. If a computation was made, or anything, even better, let's just note it down
2. do we know, or can we research (git history?) why "2" was selected in the first place?

Comment (gjkim42) [reply test/e2e_node/container_lifecycle_test.go:244]: 1. looking at the error message, it seems that there can be 1~2 seconds delay before the container start its postStart. commented it for the future reader.
2. not sure about the original decision but I guess we didn't expect it to flake. https://github.com/kubernetes/kubernetes/pull/116426 @SergeyKanzhelev

Comment (ffromani) [reply test/e2e_node/container_lifecycle_test.go:244]: thanks @gjkim42 from my POV both additions answers my questions.

**Comment by ffromani** [comment]
/lgtm
/hold
to let other reviewers chime in, feel free to remove anytime.

**Review by tzneal — APPROVED** [review]
/lgtm

But can we remove the 'Fixes 120458' from the description? This will keep it from autoclosing the flake issue until we've had time to verify that the flake is actually resolved. Its in the instructions for PRs:

> _If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`

**Comment by gjkim42** [comment]
> But can we remove the 'Fixes 120458' from the description?

Fixed it.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**